src/dune/dpath.ml

```diff
@@ -1,50 +1,83 @@
 open Stdune
 
+let install_dir = Path.Build.(relative root) "install"
+
+let alias_dir = Path.Build.(relative root ".aliases")
+
 type target_kind =
   | Regular of Context_name.t * Path.Source.t
   | Alias of Context_name.t * Path.Source.t
   | Install of Context_name.t * Path.Source.t
   | Other of Path.Build.t
 
-type path_kind =
+module Target_dir = struct
+  type context_related =
+    | Root
+    | With_context of Context_name.t * Path.Source.t
+
+  let build_dir = function
+    | Root -> Path.Build.root
+    | With_context (ctx, s) ->
+      Path.Build.append_source (Context_name.build_dir ctx) s
+
+  type t =
+    | Install of context_related
+    | Alias of context_related
+    | Regular of context_related
+    | Invalid of Path.Build.t
+
+  (* _build/foo or _build/install/foo where foo is invalid *)
+
+  let of_target ((fn : Path.Build.t) as original_fn) : t =
+    match Path.Build.extract_first_component fn with
+    | None ->
+      if Path.Build.is_root fn then
+        Regular Root
+      else
+        Invalid fn
+    | Some (name, sub) -> (
+      if name = Path.Build.basename alias_dir then
```

Comment:
It seems worth putting the result of `Path.Build.basename alias_dir` into a global variable here, otherwise we are doing a few string manipulations and allocations everytime we call this function just for an equality check.